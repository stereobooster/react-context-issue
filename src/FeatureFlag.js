// @flow
import React, { type Node } from "react";
import { Feature } from "react-feature";

export const FeatureContext = React.createContext({
  myFeature: true
});

export default ({ name, children }) => (
  <FeatureContext.Consumer>
    {config => (
      <Feature name={name} config={config}>
        {children}
      </Feature>
    )}
  </FeatureContext.Consumer>
);
