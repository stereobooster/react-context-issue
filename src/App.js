import React, { Component } from "react";
import FeatureFlag from "./FeatureFlag.js";

class App extends Component {
  render() {
    return (
      <div>
        <FeatureFlag name="myFeature">My feature is enabled</FeatureFlag>
        <FeatureFlag name="myFeature">My feature is disabled</FeatureFlag>
      </div>
    );
  }
}

export default App;
