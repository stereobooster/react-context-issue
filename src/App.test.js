import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import App from "./App";
import { FeatureContext } from "./FeatureFlag";

describe("renders", () => {
  const subject = ({ myFeature }) => (
    <div>
      <FeatureContext.Provider value={{ myFeature }}>
        <App />
      </FeatureContext.Provider>
    </div>
  );

  it("with myFeature enabled", () => {
    const tree = renderer.create(subject({ myFeature: true })).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it("with myFeature disabled", () => {
  //   const tree = renderer.create(subject({ myFeature: false })).toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  it("mounts with myFeature disabled", () => {
    const tree = mount(subject({ myFeature: false }));
    expect(true).toBe(true);
    tree.unmount();
  });

  // it("mounts with myFeature enabled", () => {
  //   const tree = mount(subject({ myFeature: true }));
  //   expect(true).toBe(true);
  //   tree.unmount();
  // });
});
