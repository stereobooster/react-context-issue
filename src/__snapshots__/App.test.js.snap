// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with myFeature enabled 1`] = `
<div>
  <div>
    My feature is enabled
    My feature is disabled
  </div>
</div>
`;
